### ISSUE TITLE
Use the ense_vector_sequence, Check failed: label.getSize() == numSamples (4 vs. 135)

### ISSUE THREAD

**Issue by hawk199** [PaddlePaddle/Paddle#10170]
# the error message:
```
I0419 12:26:53.088719 8 Util.cpp:166] commandline: --use_gpu=False --trainer_count=1
W0419 12:26:53.088758 8 CpuId.h:112] PaddlePaddle wasn't compiled to use avx instructions, but these are available o
n your machine and could speed up CPU computations via CMAKE .. -DWITH_AVX=ON
I0419 12:26:53.095015 8 GradientMachine.cpp:94] Initing parameters..
I0419 12:26:53.096026 8 GradientMachine.cpp:101] Init parameters done.
1416
F0419 12:26:53.189138 8 Matrix.cpp:3804] Check failed: label.getSize() == numSamples (4 vs. 135)
*** Check failure stack trace: ***
@ 0x7f1bfd49749d google::LogMessage::Fail()
@ 0x7f1bfd49af4c google::LogMessage::SendToLog()
@ 0x7f1bfd496fc3 google::LogMessage::Flush()
@ 0x7f1bfd49c45e google::LogMessageFatal::~LogMessageFatal()
@ 0x7f1bfd3ce971 paddle::CpuMatrix::oneHotCrossEntropy()
@ 0x7f1bfd1b4965 paddle::CostLayer::forward()
@ 0x7f1bfd1909f9 paddle::NeuralNetwork::forward()
@ 0x7f1bfd168d83 paddle::GradientMachine::forwardBackward()
@ 0x7f1bfd473999 GradientMachine::forwardBackward()
@ 0x7f1bfd1132ac _wrap_GradientMachine_forwardBackward
@ 0x4c30ce PyEval_EvalFrameEx
@ 0x4b9ab6 PyEval_EvalCodeEx
@ 0x4c1e6f PyEval_EvalFrameEx
@ 0x4b9ab6 PyEval_EvalCodeEx
@ 0x4c16e7 PyEval_EvalFrameEx
@ 0x4b9ab6 PyEval_EvalCodeEx
@ 0x4c16e7 PyEval_EvalFrameEx
@ 0x4b9ab6 PyEval_EvalCodeEx
@ 0x4c1e6f PyEval_EvalFrameEx
@ 0x4b9ab6 PyEval_EvalCodeEx
@ 0x4eb30f (unknown)
@ 0x4e5422 PyRun_FileExFlags
@ 0x4e3cd6 PyRun_SimpleFileExFlags
@ 0x493ae2 Py_Main
@ 0x7f1c16873830 __libc_start_main
@ 0x4933e9 _start
@ (nil) (unknown)
Aborted
```

# my code:

```
# -*- coding: utf-8 -*-

import os
import json
import numpy as np
import paddle.v2 as paddle

with_gpu = os.getenv('WITH_GPU', '0') != '0'


def load_train_data():
    filename="/work/footballOdds.json"
    data=load_data(filename)
    size=len(data)
    print(size)
    
    def reader():
        for i in range(size):
            lvalue=0
            if data[i][2]>data[i][3]:
                lvalue=0
            elif data[i][2]==data[i][3]:
                lvalue=1
            else:
                lvalue=2
            yield data[i][4],lvalue
    return reader

def load_test_data():
    filename="/work/footballOddsTest.json"
    data=load_data(filename)
    size=len(data)
    def reader():
        for i in range(size):
            lvalue=0
            if data[i][2]>data[i][3]:
                lvalue=0
            elif data[i][2]==data[i][3]:
                lvalue=1
            else:
                lvalue=2
            yield data[i][4],lvalue
    return reader
def row_to_array_forodds(row):
    return [row['hometeam'],
            row['guestteam'],
            row['homescore'],
            row['guestscore'],
            row['odds_data']]


def load_data(filename):
    fo = open(filename, "r")
    data=[]
    for line in fo:
        item=json.loads(line)
        data.append(row_to_array_forodds(item))
    return data


def multilayer_perceptron(match_analysis):
    # The first fully-connected layer
    hidden1 = paddle.layer.fc(input=match_analysis, size=128, act=paddle.activation.Relu())
    # The second fully-connected layer and the according activation function
    hidden2 = paddle.layer.fc(
        input=hidden1, size=64, act=paddle.activation.Relu())
    # The thrid fully-connected layer, note that the hidden size should be 10,
    # which is the number of unique digits
    predict = paddle.layer.fc(
        input=hidden2, size=10, act=paddle.activation.Softmax())
    return predict

def main():
    paddle.init(use_gpu=False, trainer_count=1)

    # define network topology
    match_analysis = paddle.layer.data(
        name='match', type=paddle.data_type.dense_vector_sequence(4))
    label = paddle.layer.data(
        name='label', type=paddle.data_type.integer_value(3))

    # Here we can build the prediction network in different ways. Please
    # choose one by uncomment corresponding line.
    # predict = softmax_regression(images)
    predict = multilayer_perceptron(match_analysis)
    #predict = convolutional_neural_network(images)

    cost = paddle.layer.classification_cost(input=predict, label=label)

    parameters = paddle.parameters.create(cost)

    optimizer = paddle.optimizer.Momentum(
        learning_rate=0.1 / 128.0,
        momentum=0.9,
        regularization=paddle.optimizer.L2Regularization(rate=0.0005 * 128))

    trainer = paddle.trainer.SGD(
        cost=cost, parameters=parameters, update_equation=optimizer)

    lists = []

    def event_handler(event):
        if isinstance(event, paddle.event.EndIteration):
            if event.batch_id % 100 == 0:
                print("Pass %d, Batch %d, Cost %f, %s" % (
                    event.pass_id, event.batch_id, event.cost, event.metrics))
        if isinstance(event, paddle.event.EndPass):
            # save parameters
            with open('params_pass_%d.tar' % event.pass_id, 'w') as f:
                trainer.save_parameter_to_tar(f)

            result = trainer.test(reader=paddle.batch(load_test_data(), batch_size=2))
            print ("Test with Pass %d, Cost %f, %s\n" % (event.pass_id, result.cost, result.metrics))
            lists.append((event.pass_id, result.cost,
                          result.metrics['classification_error_evaluator']))

    trainer.train(
        reader=paddle.batch(
            paddle.reader.shuffle(load_train_data(), buf_size=500),
            batch_size=4),
        event_handler=event_handler,
        num_passes=50)

    # find the best pass
    best = sorted(lists, key=lambda list: float(list[1]))[0]
    print('Best pass is %s, testing Avgcost is %s' % (best[0], best[1]))
    print ('The classification accuracy is %.2f%%' % (100 - float(best[2]) * 100))

    #def load_image(file):
    #    im = Image.open(file).convert('L')
    #    im = im.resize((28, 28), Image.ANTIALIAS)
    #    im = np.array(im).astype(np.float32).flatten()
    #    im = im / 255.0 * 2.0 - 1.0
    #    return im

    #test_data = []
    #cur_dir = os.path.dirname(os.path.realpath(__file__))
    #test_data.append((load_image(cur_dir + '/image/infer_3.png'), ))

    #probs = paddle.infer(
    #    output_layer=predict, parameters=parameters, input=test_data)
    #lab = np.argsort(-probs)  # probs and lab are the results of one batch data
    #print "Label of image/infer_3.png is: %d" % lab[0][0]


if __name__ == '__main__':
    main()

```

# my data:

`[[2.58, 3.25, 2.9, 2.0], [2.53, 3.25, 2.95, 27.0], [2.51, 3.25, 3.0, 47.0], [2.49, 3.25, 3.0, 55.0], [2.47, 3.3, 3.0, 120.0], [2.42, 3.3, 3.05, 152.0], [2.44, 3.3, 3.05, 212.0], [2.47, 3.3, 3.0, 243.0], [2.44, 3.3, 3.1, 283.0], [2.44, 3.2, 3.1, 910.0], [2.44, 3.15, 3.1, 1058.0], [2.42, 3.15, 3.15, 1289.0], [2.42, 3.05, 2.75, 1857.0], [2.39, 3.05, 2.8, 5735.0], [2.39, 3.2, 2.7, 6625.0], [2.35, 3.2, 2.75, 6626.0], [2.39, 3.2, 2.7, 6627.0]]`

I Run PaddlePaddle in Docker.
If anyone has any ideas, I would be very grateful!

**Comment by luotao1** [comment]
> Check failed: label.getSize() == numSamples (4 vs. 135)

这个说明标签数只有4个，但output的样本数有135个，没有对应上。

这个代码和 [02.recognize_digits](https://github.com/PaddlePaddle/book/blob/develop/02.recognize_digits/train.py)非常类似。请问：
```
    match_analysis = paddle.layer.data(
        name='match', type=paddle.data_type.dense_vector_sequence(4))
    label = paddle.layer.data(
        name='label', type=paddle.data_type.integer_value(3))
```
1) 为什么使用dense_vector_sequence，而不是dense_vector？代码里没有RNN相关layer，可以使用dense_vector。
2) label的integer_value是3类，和` predict = paddle.layer.fc(
        input=hidden2, size=10, act=paddle.activation.Softmax())`10类不匹配。

**Comment by adrianhust** [comment]
类似错误
模型配置：
```
def rnn_net(vocab_dim,
           emb_dim,
           hidden_size,
           stacked_rnn_num,
           rnn_type="lstm",
           is_infer=False):
   
    # input layers
    input = paddle.layer.data(
        name="title", type=paddle.data_type.integer_value_sequence(vocab_dim))

    # embedding layer
    input_emb = paddle.layer.embedding(input=input, size=emb_dim)

    # rnn layer
    if rnn_type == "lstm":
        for i in range(stacked_rnn_num):
            rnn_cell = paddle.networks.simple_lstm(
                input=rnn_cell if i else input_emb, size=hidden_size)
    elif rnn_type == "gru":
        for i in range(stacked_rnn_num):
            rnn_cell = paddle.networks.simple_gru(
                input=rnn_cell if i else input_emb, size=hidden_size)
    else:
        raise Exception("rnn_type error!")
    # fc(full connected) and output layer
    output = paddle.layer.fc(input=[rnn_cell],
                             size=vocab_dim,
                             act=paddle.activation.Softmax())

    if is_infer:
        last_word = paddle.layer.last_seq(input=output)
        return last_word
    else:
        return output
```
调用部分：
```
def cluster_reader_cluster(base_dir, node_id):
    def data_reader():
        paths = os.listdir(base_dir)
        for file_dir in paths:
            files = os.listdir(base_dir + '/' + file_dir)
            for fi in files:
                with open(base_dir + '/' + file_dir + '/' + fi, "r") as f:
                    for line in f:
                        items = line.strip().split('\t')
                        if len(items) < 4:
                            continue
                        title = items[0]
                        label = int(items[-1])
                        title_vec = [0] * vocab_dim
                        title_list = []
                        for it in title.decode('gb18030'):
                            if it in word_id:
                                idx = int(word_id[it])
                            else:
                                idx = 0
                            title_list.append(idx)
                        yield title_list, label

    return data_reader

def model():
    hiddensize = 128
    emb_dim = 300
    lstm_hidden = 128
    stacked_rnn_num = 2
    class_dim = 2

    lstm = rnn_conf.rnn_net(vocab_dim, emb_dim, lstm_hidden, stacked_rnn_num)
    label = paddle.layer.data("label", paddle.data_type.integer_value(class_dim))
    cost = paddle.layer.classification_cost(input=pred, label=label)
```

输入数据是一条文本，通过字典转成id，最终输入是 id 的list [id1,id5, id8  ... ]

报错：
```
I0424 21:53:06.531550  4497 GradientMachine.cpp:85] Initing parameters..
I0424 21:53:06.726786  4497 GradientMachine.cpp:92] Init parameters done.
F0424 21:53:07.103382  4497 Matrix.cpp:3277] Check failed: label.getSize() == numSamples (1 vs. 17)
*** Check failure stack trace: ***
    @     0x7f4fbb0fc27d  google::LogMessage::Fail()
    @     0x7f4fbb0ffd2c  google::LogMessage::SendToLog()
    @     0x7f4fbb0fbda3  google::LogMessage::Flush()
    @     0x7f4fbb10123e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f4fbb030875  paddle::CpuMatrix::oneHotCrossEntropy()
    @     0x7f4fbae9e3b5  paddle::CostLayer::forward()
    @     0x7f4fbaebc562  paddle::NeuralNetwork::forward()
    @     0x7f4fbaeb04c3  paddle::GradientMachine::forwardBackward()
    @     0x7f4fbb0d81c4  GradientMachine::forwardBackward()
    @     0x7f4fbad851d9  _wrap_GradientMachine_forwardBackward
    @           0x4b4cb9  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b5d10  PyEval_EvalFrameEx
    @           0x4b6b28  PyEval_EvalCodeEx
    @           0x4b6c52  PyEval_EvalCode
    @           0x4e1c7d  PyRun_FileExFlags
    @           0x4e3501  PyRun_SimpleFileExFlags
    @           0x4159dd  Py_Main
    @     0x7f4fc28c2bd5  __libc_start_main
    @           0x414b71  (unknown)
    @              (nil)  (unknown)
已放弃
```
batch size 设为了1，感觉是那一层选的类型不对，导致矩阵维度没对上

**Comment by luotao1** [comment]
可参考：https://github.com/PaddlePaddle/book/blob/develop/06.understand_sentiment/train.py#L87
@adrianhust 
```
output = paddle.layer.fc(input=[rnn_cell],
                             size=vocab_dim,
                             act=paddle.activation.Softmax())
```
fc的size要设成`class_dim`

**Comment by hawk199** [comment]
@luotao1 谢谢！参考06.understand_sentiment，采用lstm模型配置，但是我的数据是不涉及word embedding，能把 
   ```
# embedding layer
    input_emb = paddle.layer.embedding(input=input, size=emb_dim) 
```
这层去掉不？

**Comment by guoshengCS** [comment]
较久未更新，先行关闭此issue，如有需要还请重新打开。
